editions/tw5.com/tiddlers/filters/syntax/Sort Filter Run Prefix.tid

```diff
@@ -1,19 +1,25 @@
 created: 20210428083929749
-modified: 20210522162628946
-tags: [[Filter Syntax]] [[Filter Run Prefix]]
+from-version: 5.2.0
+modified: 20230319172719297
+rp-input: all titles from previous filter runs
+rp-output: the sorted result of previous filter runs
+rp-purpose: sort the input titles by the result of evaluating this filter run for each item
+rp-suffix: the `:sort` filter run prefix uses a rich suffix, see below for details
+tags: [[Named Filter Run Prefix]]
 title: Sort Filter Run Prefix
 type: text/vnd.tiddlywiki
 
-<<.from-version "5.2.0">>
-
-|''purpose'' |sort the input titles by the result of evaluating this filter run for each item |
-|''input'' |all titles from previous filter runs |
-|''suffix'' |the `:sort` filter run prefix uses a rich suffix, see below for details |
-|''output''|the sorted result of previous filter runs |
+<$railroad text="""
+\start none
+( ":sort" )
+( : ":string" | ":alphanumeric" | ":number" | ":integer" | ":version" | ":date" )
+( : ":casesensitive" /"required for string and alphanumeric"/ | ":caseinsensitive" /"required for string and alphanumeric"/  | ":reverse" /"optional"/ | - ) 
```

Comment:
Should the colon be separated, like in "Macro Call Syntax"?

Reply:
The diagram syntax is very strict (brittle).  So adding spaces may break it. So this can only be done if it can be seen and rendered.

Reply:
@pmario: I'm talking about this diagram, where the colon is separated from the value and not part of it:
![image](https://user-images.githubusercontent.com/1133745/226532470-2b36d79f-8dbc-44a6-88b9-2622ca37873d.png)

The "space" items are specific to macro calls and are not applicable here, of course.